src/test/javascript/spec/component/exam/manage/exercise-groups.component.spec.ts

```diff
@@ -0,0 +1,256 @@
+            imports: [ArtemisTestModule, TranslateTestingModule, RouterModule],
+            declarations: [
+                ExerciseGroupsComponent,
+                ExamExerciseRowButtonsComponent,
+                ProgrammingExerciseInstructorStatusComponent,
+                MockComponent(AlertComponent),
+                MockComponent(AlertErrorComponent),
+                MockComponent(BuildPlanLinkDirective),
+                MockDirective(DeleteButtonDirective),
+                MockDirective(HasAnyAuthorityDirective),
+                MockDirective(NgbTooltip),
+                MockPipe(TranslatePipe),
+            ],
+            providers: [
+                { provide: ActivatedRoute, useValue: route },
+                { provide: Router, useClass: MockRouter },
+                { provide: NgbModal, useClass: MockNgbModalService },
+            ],
+        }).compileComponents();
+
+        fixture = TestBed.createComponent(ExerciseGroupsComponent);
+        comp = fixture.componentInstance;
+
+        exerciseGroupService = TestBed.inject(ExerciseGroupService);
+        examManagementService = TestBed.inject(ExamManagementService);
+        jhiEventManager = TestBed.inject(JhiEventManager);
+        modalService = TestBed.inject(NgbModal);
+        router = TestBed.inject(Router);
+
+        groups = [
+            {
+                id: 0,
+                exercises: [
+                    { id: 3, type: ExerciseType.TEXT },
+                    { id: 4, type: ExerciseType.PROGRAMMING },
+                ],
+            } as ExerciseGroup,
+            { id: 1 } as ExerciseGroup,
+            { id: 2 } as ExerciseGroup,
+        ];
+    });
+
+    // Always initialized and bind before tests
+    beforeEach(fakeAsync(() => {
+        fixture.detectChanges();
+        tick();
+    }));
+
+    it('loads the exercise groups', fakeAsync(() => {
+        const mockResponse = new HttpResponse<Exam>({ body: exam });
+
+        spyOn(examManagementService, 'find').and.returnValue(of(mockResponse));
+
+        comp.loadExerciseGroups().subscribe((response) => {
+            expect(response.body).not.toBeNull();
+            expect(response.body!.id).toEqual(exam.id);
+        });
+
+        tick();
+    }));
+
+    it('loads exam information', fakeAsync(() => {
+        const latestIndividualEndDate = moment();
+        const mockResponse = new HttpResponse<ExamInformationDTO>({ body: { latestIndividualEndDate } });
+
+        spyOn(examManagementService, 'getLatestIndividualEndDateOfExam').and.returnValue(of(mockResponse));
+
+        comp.loadLatestIndividualEndDateOfExam().subscribe((response) => {
+            expect(response).not.toBeNull();
+            expect(response!.body).not.toBeNull();
+            expect(response!.body!.latestIndividualEndDate).toEqual(latestIndividualEndDate);
+        });
+
+        tick();
+    }));
+
+    it('removes an exercise from group', fakeAsync(() => {
+        comp.exerciseGroups = groups.slice();
```

Comment:
Why the slice? Can't you just `comp.exerciseGroups = groups`?

Reply:
Good point, forgot to remove that after assigning `groups` in `beforeEach` 👍🏻